src/Registration.js

```diff
@@ -66,13 +68,27 @@ export async function startAnyRegistrationFlow(options) {
     //     });
     //} else {
         const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
-        Modal.createTrackedDialog('Registration required', '', QuestionDialog, {
-            title: _t("Registration Required"),
-            description: _t("You need to register to do this. Would you like to register now?"),
-            button: _t("Register"),
+        const modal = Modal.createTrackedDialog('Registration required', '', QuestionDialog, {
+            hasCancelButton: true,
+            quitOnly: true,
+            title: _t("Sign In or Create Account"),
+            description: _t("Use your account or create a new one to continue."),
+            button: _t("Create Account"),
+            extraButtons: [
+                <button key="start_login"
+                onClick={() => {
+                    modal.close();
+                    dis.dispatch({action: 'start_login',
+                                  screenAfterLogin: options.screen_after});
+                               }
+                        }>
+                    { _t('Sign In') }
+                </button>,
+            ],
             onFinished: (proceed) => {
                 if (proceed) {
-                    dis.dispatch({action: 'start_registration'});
+                    dis.dispatch({action: 'start_registration',
+                                  screenAfterLogin: options.screen_after});
```

Comment:
We have line lengths to spare:
```suggestion
                    dis.dispatch({action: 'start_registration', screenAfterLogin: options.screen_after});
```